src/lumigo_tracer/lumigo_utils.py

```diff
@@ -235,9 +260,19 @@ def report_json(region: Union[None, str], msgs: List[dict], should_retry: bool =
     :return: The duration of reporting (in milliseconds),
                 or 0 if we didn't send (due to configuration or fail).
     """
-    global edge_connection
+    if not Configuration.should_report:
+        return 0
     get_logger().info(f"reporting the messages: {msgs[:10]}")
+    try:
+        prune_trace: bool = not os.environ.get("LUMIGO_PRUNE_TRACE_OFF", "").lower() == "true"
+        to_send = _create_request_body(msgs, prune_trace).encode()
+    except Exception as e:
+        get_logger().exception("Failed to create request: A span was lost.", exc_info=e)
```

Comment:
Are you sure about it? We do it in different places in the code:
![image](https://user-images.githubusercontent.com/49228804/101603804-c4276e80-3a08-11eb-89dd-b54d92214eb0.png)

![image](https://user-images.githubusercontent.com/49228804/101604281-5af42b00-3a09-11eb-8e11-93a4ad11edf5.png)

Reply:
Looks like both options are ok: (https://docs.python.org/3.7/library/logging.html)
"If exc_info does not evaluate as false, it causes exception information to be added to the logging message. If an exception tuple (in the format returned by sys.exc_info()) or an exception instance is provided, it is used; otherwise, sys.exc_info() is called to get the exception information."

Reply:
awesome, good to learn! :)